tests/unit/plugins/modules/test_proxmox_backup_info.py

```diff
@@ -0,0 +1,280 @@
+    {
+        "enabled": 1,
+        "id": "backup-83831498-c631",
+        "mailnotification": "always",
+        "mode": "snapshot",
+        "next-run": 1735138800,
+        "notes-template": "guestname",
+        "schedule": "06,18:30",
+        "storage": "local",
+        "type": "vzdump",
+        "vmid": "100"
+    },
+    {
+        "enabled": 1,
+        "id": "backup-70025700-2302",
+        "mailnotification": "always",
+        "mode": "snapshot",
+        "next-run": 1735385400,
+        "notes-template": "guestname",
+        "schedule": "sat 15:00",
+        "storage": "local",
+        "type": "vzdump",
+        "vmid": "100,101,102"
+    }
+]
+
+
+class TestProxmoxBackupInfoModule(ModuleTestCase):
+    def setUp(self):
+        super(TestProxmoxBackupInfoModule, self).setUp()
+        proxmox_utils.HAS_PROXMOXER = True
+        self.module = proxmox_backup_info
+        self.connect_mock = patch(
+            "ansible_collections.community.general.plugins.module_utils.proxmox.ProxmoxAnsible._connect",
+        ).start()
+        self.connect_mock.return_value.cluster.resources.get.return_value = (
+            RESOURCE_LIST
+        )
+        self.connect_mock.return_value.cluster.backup.get.return_value = (
+            BACKUP_JOBS
+        )
+
+    def tearDown(self):
+        self.connect_mock.stop()
+        super(TestProxmoxBackupInfoModule, self).tearDown()
+
+    def test_module_fail_when_required_args_missing(self):
+        with pytest.raises(AnsibleFailJson) as exc_info:
+            set_module_args({})
+            self.module.main()
+
+        result = exc_info.value.args[0]
+        assert result["msg"] == "missing required arguments: api_host, api_user"
+
+    def test_get_all_backups_information(self):
+        with pytest.raises(AnsibleExitJson) as exc_info:
+            set_module_args({
+                'api_host': 'proxmoxhost',
+                'api_user': 'root@pam',
+                'api_password': 'supersecret'
+            })
+            self.module.main()
+
+        result = exc_info.value.args[0]
+        assert result["backup_info"] == EXPECTED_BACKUP_OUTPUT
+
+    def test_get_specific_backup_information_by_vmname(self):
+        with pytest.raises(AnsibleExitJson) as exc_info:
+            vmname = 'test01'
+            expected_output = [
+                backup for backup in EXPECTED_BACKUP_OUTPUT if backup["vm_name"] == vmname
+            ]
+            set_module_args({
+                'api_host': 'proxmoxhost',
+                'api_user': 'root@pam',
+                'api_password': 'supersecret',
+                'vm_name': vmname
+            })
+            self.module.main()
+
+        result = exc_info.value.args[0]
+        assert result["backup_info"] == expected_output
+        assert len(result["backup_info"]) == 2
+
+    def test_get_specific_backup_information_by_vmid(self):
+        with pytest.raises(AnsibleExitJson) as exc_info:
+            vmid = "101"
```

Comment:
The `vmid` parameter is int, so this should be:
```suggestion
            vmid = 101
```

Reply:
I see you changed the param to str - so I assume you have confirmed that it _can_ be a string as well. In that case, then keep the quotes here.